app/models/application_record.rb

```diff
@@ -1,3 +1,6 @@
 class ApplicationRecord < ActiveRecord::Base
   self.abstract_class = true
+  EMAIL_REGEXP = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
```

Comment:
Unless this regex is going to be used in every model we should avoid having it here, please move it to the place where it is being used.

Reply:
Also a better option is using a gem for this validation

Reply:
ok.